state/controller.go

```diff
@@ -25,6 +29,46 @@ func controllerKey(controllerUUID string) string {
 	return fmt.Sprintf("%s#%s", controllerGlobalKey, controllerUUID)
 }
 
+// Controller encapsulates state for the Juju controller as a whole,
+// as opposed to model specific functionality.
+type Controller struct {
+	clock                  clock.Clock
+	controllerModelTag     names.ModelTag
+	controllerTag          names.ControllerTag
+	mongoInfo              *mongo.MongoInfo
+	session                *mgo.Session
+	policy                 Policy
+	newPolicy              NewPolicyFunc
+	runTransactionObserver RunTransactionObserverFunc
+}
+
+// Close the connection to the database.
+func (ctlr *Controller) Close() error {
+	return errors.Trace(ctlr.session.Close())
+}
+
+// NewState returns a new State instance for the specified model. The
+// connection uses the same credentials and policy as the Controller.
+func (ctlr *Controller) NewState(modelTag names.ModelTag) (*State, error) {
+	session := ctlr.session.Copy()
+	st, err := newState(
+		modelTag,
+		ctlr.controllerModelTag,
+		session,
+		ctlr.mongoInfo,
+		ctlr.newPolicy,
+		ctlr.clock,
+		ctlr.runTransactionObserver,
+	)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if err := st.start(ctlr.controllerTag); err != nil {
```

Comment:
Actually, this didn't work out so I've backed it out. Given it's not that important and I don't want to lose momentum I'm going to skip this.